src/classes/GE_GiftEntryController.cls

```diff
@@ -173,6 +178,20 @@ public with sharing class GE_GiftEntryController {
         }
     }
 
+    /**
+     * @description Validates that the currency of a batch matches the currently logged in
+     * user's currency
+     * @param batchId : ID of the NPSP_Data_Import_Batch__c
+     */
+    private static void validateCurrencies (Id batchId) {
+        String batchCurrencyIsoCode = getBatchCurrencyIsoCode(batchId);
+        if (batchCurrencyIsoCode != UserInfo.getDefaultCurrency()) {
+           throw new CurrencyMismatchException(String.format(
```

Comment:
I see that we are throwing an exception for mismtached ISO code here. Are we doing this as well as the warning message in the UI? Assumed that this exception would also override the warning message.